File: agent/modules/Generics/WebScreenshot.py
class WebScreenshot:
    def __init__(self, serviceId, path, picture):
        self.serviceId = serviceId
        self.path = path
        self.picture = picture

    def toDict(self):
        return {
            "type": "__webscreenshot__",
            "service_id": self.serviceId,
            "path": self.path,
            "picture": self.picture            
        }

    def __repr__(self):
        return str(self.toDict())

File: server/app/Agent/Models/WebScreenshot.php
<?php


namespace App\Agent\Models;

use Storage;

class WebScreenshot
{

    public $service_id = null;
    public $picture = null;
    public $type = '__webscreenshot__';

    public function __construct($service_id, $picture)
    {
        $this->service_id = $service_id;
        $this->picture = $picture;
    }

    public static function  fromEloquent(\App\WebScreenshot $webScreenshot) {
        $file = Storage::disk('local')->get($webScreenshot->image_name);
        return new WebScreenshot($webScreenshot->image_name, $file);
    }

    public function toEloquent() {
        $new = new \App\WebScreenshot();
        $new->service_id = $this->service_id;
        $new->image_name = $this->service_id . '.jpeg';
        return $new;
    }

}